srcs/http/response/HttpBaseResponse.cpp

```diff
@@ -26,13 +26,13 @@ std::string	 HttpBaseResponse::Content() const {
 	return raw_response_;
 }
 
-void	HttpBaseResponse::Serve_(File file) {
+void	HttpBaseResponse::Serve_(File file, int error_code_) {
 	HttpResponse::HeadersMap headers;
 	std::string body;
 
 	headers.insert(std::make_pair("Content-Type", file.GetMimeType()));
 	body = file.GetContent();
-	SetRawResponse_(200, headers, body);
+	SetRawResponse_(error_code_, headers, body);
```

Comment:
the underscore in the name of the parameters makes it look like the error_code_ is a private variable of the HttpBaseResponse class. (SPOILER, it isn't).
On the other hand, again, this thing with passing objects by copy... is this one of those instances where we actually don't have an overhead when doing so because we need a copy of the object in this particular scope?

Reply:
Good catch :+1:, I've changed it. Now we use the private `error_code_` variable.

We could pass most of the parameters as const references, but those changes are outside of the scope of this PR.

Reply:
Would not be more comprehensive to have something like:

1. Instead of having an `error_code_` -> call it just `code_` because requests have codes that are errors or not
2. In line 17 of this same class, instead of assigning a `0` value, we can assign a `200` by default. This means no errors for now.
3. In the specialized classes  constructor (such as [GET](https://github.com/42AGV/webserv/blob/6c5c1851188b75fe368f139b7e6130217d395873/srcs/http/response/get/HttpGetResponse.cpp#L7), POST, DELETE, rediection etc) where we check for errors at the beggining, instead of doing something like `if (error_code_)...` we could instead, update the logic to `if (code != 200)...`

Reply:
@vdedios I like this, and it makes more sense.